ncm-spma/src/test/perl/apt-cleanup-sources.t

```diff
@@ -0,0 +1,117 @@
+# -*- mode: cperl -*-
+# ${license-info}
+# ${author-info}
+# ${build-info}
+
+=pod
+
+=head1 DESCRIPTION
+
+Tests for the cleanup_old_sources method.  This method ensures that
+outdated sources in /etc/apt/sources.list.d/ are removed.
+
+=head1 TESTS
+
+=cut
+
+use strict;
+use warnings;
+use Readonly;
+use English;
+
+use Test::Quattor;
+use Test::More;
+use Test::MockModule;
+use NCM::Component::spma::apt;
+use CAF::Path;
+use CAF::FileWriter;
+use Set::Scalar;
+
+
+Readonly my $SOURCE_DIR => "target/test/clean.sources.list.d";
+
+my $sources = [{
+    name => "source1",
+    owner => 'localuser@localdomain',
+    protocols => [{
+        name => "http",
+        url => "http://localhost.localdomain",
+    }],
+}];
+
+my $cmp = NCM::Component::spma::apt->new("spma");
+my $mock = Test::MockModule->new('NCM::Component::spma::apt');
+
+my $immutable;
+my $directory_exists;
+my $sourcefiles;
+
+
+$mock->mock('_glob', sub {
+    return @$sourcefiles;
+});
+
+
+$mock->mock('cleanup', sub {
```

Comment:
whole of CAF::Path is already mocked and integrated in Test::Quattor 1.54 (even the not yet merged CAF::Path listdir)